controller/api/destination/server.go

```diff
@@ -364,17 +364,44 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 	return nil
 }
 
-func (s *server) createAddress(pod *corev1.Pod, port uint32) (watcher.Address, error) {
+// getPortForPod returns the port that a `pod` is listening on.
+//
+// If the `host` address given is the HostIP, use the HostPort mapping for the `pod`
+// to determine the port to return.
+func (s *server) getPortForPod(pod *corev1.Pod, host string, port uint32) (uint32, error) {
```

Comment:
`host` seems a little ambiguous. `ip` might be clearer. or `targetIP`?

In fact, is it possible to deal with an `IPAddress` type directly?

Reply:
So I would have to immediately turn it back into a string but I agree a little Postel's law would go a long way here so now I'm parsing it, checking for an error, and using the string in the comparisons.